emukit/quadrature/kernels/quadrature_matern32.py

```diff
@@ -0,0 +1,170 @@
+    """Augments a ProductMatern32 kernel with integrability."""
+
+    def __init__(
+        self,
+        matern_kernel: IProductMatern32,
+        integral_bounds: Optional[List[Tuple[float, float]]],
+        measure: Optional[IntegrationMeasure],
+        variable_names: str = "",
+    ) -> None:
+        """
+        :param matern_kernel: Standard emukit matern32-kernel.
+        :param integral_bounds: Defines the domain of the integral. List of D tuples, where D is the dimensionality
+        of the integral and the tuples contain the lower and upper bounds of the integral.
+        i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]. None for infinite bounds
+        :param measure: An integration measure. None means the standard Lebesgue measure is used.
+        :param variable_names: The (variable) name(s) of the integral
+        """
+
+        super().__init__(
+            kern=matern_kernel, integral_bounds=integral_bounds, measure=measure, variable_names=variable_names
+        )
+
+    @property
+    def lengthscales(self):
+        return self.kern.lengthscales
+
+    @property
+    def variance(self):
+        return self.kern.variance
+
+    def qK(self, x2: np.ndarray) -> np.ndarray:
+        """ProductMatern32 kernel with the first component integrated out aka kernel mean
+
+        :param x2: remaining argument of the once integrated kernel, shape (n_point N, input_dim)
+        :returns: kernel mean at location x2, shape (1, N)
+        """
+        raise NotImplementedError
+
+    def Kq(self, x1: np.ndarray) -> np.ndarray:
+        """ProductMatern32 kernel with the second component integrated out aka kernel mean
+
+        :param x1: remaining argument of the once integrated kernel, shape (n_point N, input_dim)
+        :returns: kernel mean at location x1, shape (N, 1)
+        """
+        return self.qK(x1).T
+
+    def qKq(self) -> float:
+        """ProductMatern32 kernel integrated over both arguments x1 and x2.
+
+        :returns: double integrated kernel
+        """
+        raise NotImplementedError
+
+    def dqK_dx(self, x2: np.ndarray) -> np.ndarray:
+        """
+        gradient of the kernel mean (integrated in first argument) evaluated at x2
+        :param x2: points at which to evaluate, shape (n_point N, input_dim)
+        :return: the gradient with shape (input_dim, N)
+        """
+        raise NotImplementedError
+
+    def dKq_dx(self, x1: np.ndarray) -> np.ndarray:
+        """
+        gradient of the kernel mean (integrated in second argument) evaluated at x1
+        :param x1: points at which to evaluate, shape (n_point N, input_dim)
+        :return: the gradient with shape (N, input_dim)
+        """
+        return self.dqK_dx(x1).T
+
+
+class QuadratureProductMatern32LebesgueMeasure(QuadratureMatern32):
+    """A Matern32 product kernel with integrability over the standard Lebesgue measure.
+
+    Can only be used with finite integral bounds.
```

Comment:
This needs to be validated somewhere

Reply:
Yes, indeed. This needs to be checked for all kernel embeddings that use the Lebesgue measure actually (and it isn't so far). 

More generally I may re-do how we handle the Lebesgue integration measure at some point. So far it is implied by `measure=None` which leads annoying if-else statements in the code.